book_haven_api/views.py

```diff
@@ -24,12 +26,38 @@ def get_a_library(request, library_id):
             "detail": "The library with id " + str(library_id) + " does not exist."
         }]},status=status.HTTP_404_NOT_FOUND)
     
-@api_view(['GET'])
+@api_view(['GET', 'POST'])
 def get_library_books(request, library_id):
-    library = Library.objects.get(id=library_id)
-    books = library.book_set
-    serializer = BookSerializer(books, many=True)
-    return Response({ "data": serializer.data }, status=status.HTTP_200_OK)
+    if request.method == 'GET':
+        library = Library.objects.get(id=library_id)
+        books = library.book_set
+        serializer = BookSerializer(books, many=True)
+        return Response({ "data": serializer.data }, status=status.HTTP_200_OK)
+    
+    elif request.method == 'POST':
```

Comment:
You actually shouldn't need this `elif`. Generally if the previous branch of the logic has an explicit `return`, you can assume that if you make it here it's the other branch. As this method only accepts `GET` and `POST`, if the `if` doesn't fire then you can be fairly certain you have a `POST` request. Not always a bad idea to check, but in this case you can remove the `elif` and then check if you want.